test/end2end_test.go

```diff
@@ -5081,34 +5081,64 @@ func (s) TestClientForwardsGrpcAcceptEncodingHeader(t *testing.T) {
 	}
 }
 
+// wrapCompressor is a wrapper of encoding.Compressor which maintains count of
+// Compressor method invokes.
+type wrapCompressor struct {
+	encoding.Compressor
+	compressInvokes int32
+}
+
+func (wc *wrapCompressor) Compress(w io.Writer) (io.WriteCloser, error) {
+	atomic.AddInt32(&wc.compressInvokes, 1)
+	return wc.Compressor.Compress(w)
+}
+
+func setupGzipWrapCompressor(t *testing.T) *wrapCompressor {
+	oldC := encoding.GetCompressor("gzip")
+	c := &wrapCompressor{Compressor: oldC}
+	encoding.RegisterCompressor(c)
+	t.Cleanup(func() {
+		encoding.RegisterCompressor(oldC)
+	})
+	return c
+}
+
 func (s) TestSetSendCompressorSuccess(t *testing.T) {
 	for _, tt := range []struct {
-		desc          string
-		resCompressor string
-		wantErr       error
+		name                string
+		desc                string
+		dialOpts            []grpc.DialOption
+		resCompressor       string
+		wantCompressInvokes int32
 	}{
 		{
-			desc:          "gzip_response_compressor",
-			resCompressor: "gzip",
+			name:                "identity_request_and_gzip_response",
+			desc:                "request is uncompressed and response is gzip compressed",
+			resCompressor:       "gzip",
+			wantCompressInvokes: 1,
 		},
 		{
-			desc:          "identity_response_compressor",
-			resCompressor: "identity",
+			name:                "gzip_request_and_identity_response",
+			desc:                "request is gzip compressed and response is uncompressed with identity",
+			resCompressor:       "identity",
+			dialOpts:            []grpc.DialOption{grpc.WithCompressor(grpc.NewGZIPCompressor())},
```

Comment:
I believe this should be `grpc.UseCompressor(gzip.Name)` instead.

Reply:
Passing `grpc.UseCompressor` would invoke `Compress` on the wrapCompressor for request compression as well - we can't clearly find from wrapCompressor if invokes were made for request or response. Passing `grpc.WithCompressor` ensures request flow uses different instance of compressor, and wrapCompressor is only used only in the response flow.

Reply:
Ohhh, good point.  This is because we don't have a way to specify client/server-local compressors or codecs, like we do have the ability to do for [name resolvers](https://pkg.go.dev/google.golang.org/grpc#WithResolvers).  We probably should think about adding that, but in the meantime, would you mind leaving a comment on this line about why we're using this deprecated function?  E.g. `// Use WithCompressor instead of UseCompressor to avoid counting the client's compressor usage.` or something like that.

Thanks!